AGENTS.md

```diff
@@ -0,0 +1,924 @@
+- Location: `tests/foreman/cli/`
+- Example: `tests/foreman/cli/test_activationkey.py`
+
+**API Tests**: Use Nailgun to interact with Satellite API
+- Location: `tests/foreman/api/`
+- Example: `tests/foreman/api/test_activationkey.py`
+
+**Upgrade Tests**: Uses`SharedResource` for single-test upgrade scenarios:
+- Location: 'tests/new_upgrades'
+- Example: 'tests'new_upgrades/test_activation_key.py'
+
+---
+
+## Architecture
+
+Robottelo follows a **layered testing architecture** that separates test logic, fixtures, and helper utilities:
+
+### Layer 1: Test Layer
+The top layer where actual test functions are written using pytest.
+
+- **Purpose**: Define test scenarios and assertions
+- **Location**: `tests/foreman/{api,cli,ui}/`
+- **Example**: `def test_positive_create_activation_key(...)`
+- **Responsibilities**:
+  - Execute test steps
+  - Assert expected outcomes
+  - Use fixtures for setup/teardown
+
+### Layer 2: Fixture Layer
+The middle layer providing reusable test setup and teardown logic.
+
+- **Purpose**: Provide test dependencies and resources
+- **Location**: `pytest_fixtures/`
+- **Types**:
+  - **Core fixtures**: `pytest_fixtures/core/` (Satellite, Broker, ContentHost)
+  - **Component fixtures**: `pytest_fixtures/component/` (per-feature fixtures)
+- **Responsibilities**:
+  - Provision infrastructure (Satellite, hosts)
+  - Configure test prerequisites
+  - Clean up resources after tests if needed
+
+### Layer 3: Helper/Utility Layer
+The bottom layer containing helper classes, utilities, and base implementations.
+
+- **Purpose**: Provide reusable code for common operations
+- **Location**: `robottelo/`
+- **Components**:
+  - **API helpers**: `robottelo/api/` (Nailgun entities)
+  - **Host classes**:  `robottelo.hosts.py` (Base functionality for ContentHost, Capsule, Satellite interaction s)
+  - **CLI helpers**: `robottelo/cli/` (Hammer command wrappers)
+  - **Host helpers**: `robottelo/host_helpers/` (Satellite/ContentHost mixins)
+  - **Utilities**: `robottelo/utils/` (decorators, data factories, etc.)
+
+### Layer 4: Infrastructure Layer
+External services and tools that tests depend on.
+
+- **Broker**: VM/Container host provisioning
+- **Manifester**: Subscription manifest generation
+- **Report Portal**: Test result reporting
+- **Vault**: Secret management
+
+---
+
+## Key Concepts
+
+### 1. **Fixtures**
+
+Pytest fixtures provide test dependencies and setup/teardown logic.
+
+**Core Fixtures**:
+- `target_sat`: A Satellite instance for the test
+- `module_target_sat`: Module-scoped Satellite instance
+- `function_sca_manifest`: Function-scoped SCA manifest
+- `rhel_contenthost`: RHEL content host for testing
+
+**Component Fixtures**:
+- `module_ak_with_cv`: Activation key with content view
+- `module_lce`: Lifecycle environment
+- `module_org`: Organization
+
+**Fixture Scopes**:
+- `function`: Per test function (default)
+- `module`: Per test module
+- `session`: Per test session
+- `class`: Per test session
```

Comment:
```suggestion
- `class`: Per test class
```
Also, `class` should fit between `function` and `module` if we want to order the particular types per their scope.